### PR TITLE
Adding a WDL demo

### PR DESCRIPTION
**Pull request by briandoconnor**
I added a WDL workflow and instructions on how to call this tool via WDL.  I wanted to test that this tool is callable via Cromwell locally since I thought it would help with testing on the Broad's side.

I found that the ENTRYPOINT in Frank's Docker wasn't compatible with the way Cromwell calls the tool.  It seems to assume the entrypoint is bash and pipes in the command.  

To make this concrete, here's how Cromwell executes a command in a given Docker container:

```
"/bin/bash" "-c" "cat cromwell-executions/wf/ac99b2c7-2204-4827-a974-3873c3e6b5c5/call-adam/script | docker run --rm -v /Users/boconnor/Development/gitroot/BD2KGenomics/cgl-docker-lib/adam-pipeline/cromwell-executions/wf/ac99b2c7-2204-4827-a974-3873c3e6b5c5/call-adam:/root/wf/ac99b2c7-2204-4827-a974-3873c3e6b5c5/call-adam -i quay.io/ucsc_cgl/adam-pipeline /bin/bash <&0"
```

Feel free to tweak as needed or discard if I'm not understanding the issue properly.

I don't think I'm handling the output file correctly since it's not copied back to the output path specified in the JSON.  I leave that for the Broad folks to comment on/fix.

### CODE REVIEW

File: adam-pipeline/workflow.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "wf.adam.in": "/Users/boconnor/Development/gitroot/BD2KGenomics/cgl-docker-lib/adam-pipeline/test/small.sam",
```
Comment (hannes-ucsc) [inline adam-pipeline/workflow.json]: Are relative paths an option?

File: adam-pipeline/workflow.wdl [hunk 2]
```diff
@@ -0,0 +1,22 @@
+task adam {
+  File in
+  File knownSites
+  String mem
+
+  command {
+    python /opt/adam-pipeline/wrapper.py --known-sites ${knownSites} --sample ${in} --output `pwd`/small.processed.bam --memory ${mem}
```
Comment (hannes-ucsc) [inline adam-pipeline/workflow.wdl:7]: Don't like how the location of the wrapper leaks into the workflows.

File: adam-pipeline/workflow.json (hunk 1 above)
Comment (briandoconnor) [inline adam-pipeline/workflow.json]: Didn't work for me but I'm a total WDL newbie.

File: adam-pipeline/workflow.wdl (hunk 2 above)
Comment (fnothaft) [inline adam-pipeline/workflow.wdl:7]: +1, that's... aesthetically displeasing.

**Comment by fnothaft** [comment]
@briandoconnor I just opened #154 that has some cleanup on top of this. Can you take a look and let me know what you think?

**Comment by fnothaft** [comment]
Also, this `/bin/bash -c` business is ugly.

File: adam-pipeline/workflow.wdl (hunk 2 above)
Comment (briandoconnor) [inline adam-pipeline/workflow.wdl:7]: Agreed.  But I do like that the WDL then hides this detail from me.  I can just focus on inputs/outputs.  The key would be to make sure to clearly associate a release of the Docker image with the correct release of the WDL... if only there was a site that associated Docker images with WDL/CWL descriptors :-)

**Comment by briandoconnor** [comment]
Not sure what the right thing to do with the entrypoint is.  I think have a script/tool as the entrypoint is going to interfere with assumptions workflow systems like WDL and CWL have made about how they can execute a command inside a container.  I don't think I have a clean solution.  But this is just a demo.  Something to think about for future containers since I don't want UCSC to make hundreds of tool Docker containers with customized entrypoints and then have those be unrunnable in any other system.  It would be a lost opportunity.

**Comment by briandoconnor** [comment]
Frank, by the way, please do what you want with this pull request.  Since it's really a demo for the Broad it's your call how you want to tweak it/merge it with your changes.

**Comment by fnothaft** [comment]
@hannes-ucsc this is good to merge. Let me know if you want the merge commit cleaned up.

**Comment by fnothaft** [comment]
Tagging `needs work` again; @briandoconnor just ran into a test failure on his side.

**Comment by fnothaft** [comment]
@briandoconnor I want to remove the merge commit and push some test cleanup to this branch. Is it OK with you if I force push to this branch?

**Comment by fnothaft** [comment]
@hannes-ucsc I've removed the `needs work` tag. Let me know if you want this squashed, otherwise let's merge.

**Comment by fnothaft** [comment]
Merged, as per discussion with @hannes-ucsc! Thanks @briandoconnor.

**Pull request merged**